### PR TITLE
Implement support for reading Opal card (Sydney, Australia)

### PR DESCRIPTION
**Pull request by micolous**
# What's new

- Adds support for reading Opal card (Sydney, Australia): [format reference](https://github.com/metrodroid/metrodroid/wiki/Opal)
- Adds hooks for parsing MIFARE DESFire cards, based on MIFARE Classic and Ultralight readers. Unlike the others, it doesn't use `.verify` or `.read`, so we have a stub `stub_parser_verify_read` for that.
- Adds `mf_df_get_application` and `mf_df_get_file` helpers for parsing DESFire cards.
- Adds `furi_hal_rtc_get_days_per_month`, `furi_hal_rtc_get_days_per_year`, `furi_hal_rtc_is_leap_year` to public API to assist with date parsing.
- Corrects capitalisation of "DESFire".

# Verification 

- Scan Opal cards, should show correct balance and last activity
- Open a previous scan of an Opal card, should show up in the `Info` screen
- Scan a non-Opal DESFire card, that should not show anything about Opal
- Open a previous scan of a non-Opal DESFire card, that should also not show anything about Opal
- Repeat tests with different date / time formatting settings, that should follow system settings
- Check through different values that exercise formatting edge cases

Example dump of some slightly modified transit litter: [OpalLitter.nfc.txt](https://github.com/flipperdevices/flipperzero-firmware/files/11523840/OpalLitter.nfc.txt)

* Card number: `3085 2200 7856 2242`
* Balance: $-1.82
* Has automatic top-up enabled (modified)
* Weekly trips: 2
* Transaction number: 21
* Last transaction timestamp: 2015-07-27 at 18:24
* Last transaction mode: Rail / Metro
* Last transaction type: Tap off (distance fare)
* Checksum probably doesn't match due to modifications, but this doesn't check it.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: lib/nfc/parsers/nfc_supported_card.c [hunk 1]
```diff
@@ -65,3 +74,9 @@ bool nfc_supported_card_verify_and_parse(NfcDeviceData* dev_data) {
 
     return card_parsed;
 }
+
+bool stub_parser_verify_read(
+    NfcWorker* nfc_worker __attribute__((unused)),
+    FuriHalNfcTxRxContext* tx_rx __attribute__((unused))) {
```
Comment (gornekich) [inline lib/nfc/parsers/nfc_supported_card.c]: Use 
UNUSED(nfc_worker);
UNUSED(tx_rx);

File: firmware/targets/furi_hal_include/furi_hal_rtc.h [hunk 2]
```diff
@@ -12,6 +12,17 @@
...
+#define FURI_HAL_RTC_SECONDS_PER_HOUR (FURI_HAL_RTC_SECONDS_PER_MINUTE * 60)
+#define FURI_HAL_RTC_SECONDS_PER_DAY (FURI_HAL_RTC_SECONDS_PER_HOUR * 24)
+#define FURI_HAL_RTC_EPOCH_START_YEAR 1970
+#define FURI_HAL_RTC_IS_LEAP_YEAR(year) \
+    ((((year) % 4 == 0) && ((year) % 100 != 0)) || ((year) % 400 == 0))
+
+#define FURI_HAL_RTC_MONTHS_COUNT 12
+
+extern const uint8_t furi_hal_rtc_days_per_month[2][FURI_HAL_RTC_MONTHS_COUNT];
+
```
Comment (gornekich) [inline firmware/targets/furi_hal_include/furi_hal_rtc.h]: I wouldn't move it here.
Better to add some functions in API:

```
bool furi_hal_rtc_is_leap_year();
uint8_t furi_hal_rtc_get_days_per_months()
```

@skotopes  what do you think?

**Review by gornekich — DISMISSED** [review]

File: lib/nfc/parsers/nfc_supported_card.c (hunk 1 above)
Comment (micolous) [reply lib/nfc/parsers/nfc_supported_card.c]: Done

File: firmware/targets/furi_hal_include/furi_hal_rtc.h (hunk 2 above)
Comment (micolous) [reply firmware/targets/furi_hal_include/furi_hal_rtc.h]: Done

**Review by skotopes — APPROVED** [review]
